temporal-sdk/src/main/java/io/temporal/activity/ActivityExecutionContext.java

```diff
@@ -147,4 +147,7 @@ public interface ActivityExecutionContext {
    * an activity.
    */
   WorkflowClient getWorkflowClient();
+
+  /** Get the currently running activity instance. */
+  Object getInstance();
```

Comment:
General question, is this available in the main execute-activity inbound interceptor? Meaning there's no case where it is null (and Java has no concept of accessing the execution context from inside activity class constructors since users invoke those I assume).

Reply:
Correct